src/supermarket/app/controllers/cookbooks_controller.rb

```diff
@@ -308,4 +309,17 @@ def apply_filters
       @cookbooks = @cookbooks.filter_badges(params[:badges])
     end
   end
+
+  def classify_metrics(metric_results)
```

Comment:
```diff
diff --git a/src/supermarket/app/models/cookbook_version.rb b/src/supermarket/app/models/cookbook_version.rb
index 9123e47..40f6e97 100644
--- a/src/supermarket/app/models/cookbook_version.rb
+++ b/src/supermarket/app/models/cookbook_version.rb
@@ -6,7 +6,7 @@ class CookbookVersion < ActiveRecord::Base
   has_many :cookbook_version_platforms
   has_many :supported_platforms, through: :cookbook_version_platforms
   has_many :cookbook_dependencies, dependent: :destroy
-  has_many :metric_results
+  has_many :metric_results, -> { includes :quality_metric }

   belongs_to :cookbook
   belongs_to :user
diff --git a/src/supermarket/app/models/metric_result.rb b/src/supermarket/app/models/metric_result.rb
index c9bfbf4..8c78cff 100644
--- a/src/supermarket/app/models/metric_result.rb
+++ b/src/supermarket/app/models/metric_result.rb
@@ -1,4 +1,10 @@
 class MetricResult < ActiveRecord::Base
   belongs_to :cookbook_version
   belongs_to :quality_metric
+
+  scope :open, -> { joins(:quality_metric).where('quality_metrics.admin_only IS NULL')}
+  scope :admin_only, -> { joins(:quality_metric).where('quality_metrics.admin_only = true')}
+
+  delegate :name, to: :quality_metric
+  delegate :admin_only?, to: :quality_metric
 end
```

The above additions to the classes let me tinker on the console with:

```
# "open" because ActiveRecord::Base already has a "public" method
> cookbook_version.metric_results.open
> cookbook_version.metric_results.admin_only
# can still get the hash to get all results at once, but in open/admin buckets
> cookbook_version.metric_results.group_by(&:admin_only?)
```

And then the controllers could do something like ...

```diff
diff --git a/src/supermarket/app/controllers/cookbook_versions_controller.rb b/src/supermarket/app/controllers/cookbook_versions_controller.rb
index 839b453..be847ff 100644
--- a/src/supermarket/app/controllers/cookbook_versions_controller.rb
+++ b/src/supermarket/app/controllers/cookbook_versions_controller.rb
@@ -26,10 +26,8 @@ class CookbookVersionsController < ApplicationController
     @supported_platforms = @version.supported_platforms
     @owner_collaborator = Collaborator.new resourceable: @cookbook, user: @owner

-    @public_metric_results = []
-    @admin_metric_results = []
-
-    classify_metrics(@version.metric_results)
+    @public_metric_results = @version.metric_results.open
+    @admin_metric_results = @version.metric_results.admin_only
   end

   private
@@ -38,14 +36,4 @@ class CookbookVersionsController < ApplicationController
     @cookbook = Cookbook.with_name(params[:cookbook_id]).first!
     @version = @cookbook.get_version!(params[:version])
   end
-
-  def classify_metrics(metric_results)
-    metric_results.each do |result|
-      if result.quality_metric.admin_only?
-        @admin_metric_results << result
-      else
-        @public_metric_results << result
-      end
-    end
-  end
 end
diff --git a/src/supermarket/app/controllers/cookbooks_controller.rb b/src/supermarket/app/controllers/cookbooks_controller.rb
index 7da5c82..919c146 100644
--- a/src/supermarket/app/controllers/cookbooks_controller.rb
+++ b/src/supermarket/app/controllers/cookbooks_controller.rb
@@ -86,7 +86,8 @@ class CookbooksController < ApplicationController
     @collaborators = @cookbook.collaborators
     @supported_platforms = @cookbook.supported_platforms

-    classify_metrics(@cookbook.latest_cookbook_version.metric_results.includes(:quality_metric))
+    @public_metric_results = @latest_version.metric_results.open
+    @admin_metric_results = @latest_version.metric_results.admin_only

     respond_to do |format|
       format.atom
@@ -309,16 +310,4 @@ class CookbooksController < ApplicationController
       @cookbooks = @cookbooks.filter_badges(params[:badges])
     end
   end
-
-  def classify_metrics(metric_results)
-    @public_metric_results = []
-    @admin_metric_results = []
-    metric_results.each do |result|
-      if result.quality_metric.admin_only?
-        @admin_metric_results << result
-      else
-        @public_metric_results << result
-      end
-    end
-  end
end
```